### PR TITLE
Fixes #33669 - apply PF4 classname to all tables

### PR DESCRIPTION
**Pull request by Ron-Lavi**

### CODE REVIEW

**Comment by tbrisker** [comment]
@laviro can you add some screenshots of before and after please?

**Comment by Ron-Lavi** [comment]
New ones:

Bookmarks
gnome-shell-screenshot-ETD6A1


Architectures
gnome-shell-screenshot-AI3AB1

Hosts
Screenshot-20211011215444-1918x760

**Comment by Ron-Lavi** [comment]
Old ones

Bookmarks
gnome-shell-screenshot-2LOFB1

Architectures
gnome-shell-screenshot-ZGP6A1

Hosts
gnome-shell-screenshot-QHB9A1

**Comment by MariSvirik** [comment]
@laviro use [compact tables](https://www.patternfly.org/v4/components/table/design-guidelines#compact-spacing).

**Comment by Ron-Lavi** [comment]
Updated the screenshots in https://github.com/theforeman/foreman/pull/8828#issuecomment-939976774
using the compact table now :)

**Comment by ekohl** [comment]
That's much better but I think the striping in the current tables is very useful. Isn't there a PF4 equivalent?

Edit: also the host list column widths look worse. How does it behave on smaller screens? And if you have long hostnames?

**Comment by MariSvirik** [comment]
@ekohl - no stripping equivalent in PF4 as far as I know.
Headers should be legible in full screen and column content should be left-aligned (Recommendation column needs these fixes). Somehow there is too much free space in the Host table on the left hand side (checkbox, on/off, hostname)

If the hostname is too long, you should split it into two lines.
Example from Inventory (console.redhat) in a smaller screen 
Screenshot 2021-10-12 at 13 47 03

**Comment by Ron-Lavi** [comment]
> That's much better but I think the striping in the current tables is very useful. Isn't there a PF4 equivalent?

I didn't see one in https://patternfly-react.surge.sh/components/table
@MariSvirik is there a plan to add a striped table in PF?



> Edit: also the host list column widths look worse. How does it behave on smaller screens? And if you have long hostnames?

one of the widths was a plugin column that was added which looks OK on the original table, need a bit of adjustment now, for the original table this is how it looks with Foreman only on a laptop resolution:
gnome-shell-screenshot-ZO4EB1

this is how it looks on half of a laptop screen:
gnome-shell-screenshot-6XVGB1

**Comment by ekohl** [comment]
IMHO it's a worse UX. Overall I really dislike how much whitespace PF4 has and think the font is worse. Also, to me multiple lines feels worse than trimming the field. Of course there are still other things we can apply to be even smarter (the OS can be reduced to `<icon> <version>` instead of <icon> <name> <version>` on small screens) but we never did those so I can't hold that against PF4.

@MariSvirik what is the rationale for not having striping?

**Comment by Ron-Lavi** [comment]
@ekohl we are still using a mix of styles but at least we are moving into a more unified look of the app, next step the table PF4 component will get into Foreman and with an index page template or something alike we could reuse it all over,

for truncating the text I believe we could use a classname like `ellipsis` called `pf-m-truncate` as seen in https://patternfly-react.surge.sh/components/table/#composable-modifiers-with-table-text

honestly, I expected our `ellipsis` to just work, I will take a deeper look at it...

**Comment by MariSvirik** [comment]
@laviro @ekohl 
PF4 team got rid of them in the first place 'because it introduces too much visual noise and the team was striving for a more minimal appearance.'  
I investigated  Patternfly roadmap and found out that even though stripped tables were in a roadmap (later in the roadmap's backlog) they have been recently removed entirely. PF4 team said it's worth considering and it would be 'nice to have it as an alternative' and that it was a part of longer term visual refresh but as it was removed from the roadmap, and there is no timeframe for that, I don't expect to have it anytime soon. 

Regarding hostname length, I would rather break the hostname into two lines, so it's going to be always legible, no matter how wide the screen is (that's how console.redhat has done it). When in doubt, consult the console.redhat implementation.

Tbh, I'm not sure why there is so much white space on the table.

**Comment by Ron-Lavi** [comment]
For a striped table, we can add a CSS rule like:
```css
tr:nth-child(odd) {
  background-color: color;
}
```
gnome-shell-screenshot-CF3FB1

When removing Foreman's center-align class name from the table rows and removing hard-coded width size to each col,
the hosts' table looks much better:

Screenshot-20211014085931-1917x851

File: webpack/assets/javascripts/react_app/components/Pagination/Pagination.js [hunk 1]
```diff
@@ -49,6 +54,35 @@ const Pagination = props => {
...
+    page: __('page'),
+    itemsPerPage: __('Items per page'),
+    perPageSuffix: __('per page'),
+    toFirstPage: __('Go to first page'),
+    toPreviousPage: __('Go to previous page'),
+    toLastPage: __('Go to last page'),
+    toNextPage: __('Go to next page'),
+    optionsToggle: __('Items per page'),
+    currPage: __('Current page'),
+    paginationTitle: __('Pagination'),
```
Comment (MariaAga) [inline webpack/assets/javascripts/react_app/components/Pagination/Pagination.js:68]: Can you add `ofWord: __('of'),`?

**Comment by MariSvirik** [comment]
Hey guys, let's just follow PF4 patterns 😄. I showed those tables to some users previously (months ago) and none of them articulated need for striped tables. On the other hand, I welcome your initiative @laviro. It's good to know, that it's an easy and quick fix to change it.

**Review by tbrisker — CHANGES_REQUESTED** [review]
Thanks for opening this PR to enable the discussion! 
However, I don't think this should go in so close to the branching date and while the discussion regarding the RFC is still ongoing. There are multiple tables whose layout will require manual adjustment to look correct and other concerns that have been raised. All plugins will also need to be update to avoid mismatching tables, and possibly in some cases where we still use deface to modify tables the deface definition might also need to change.
Additionally, I would much rather get the new index template that implements actual PF4 tables completely with all the benefits of switching them to React, rather than faking the styling of our existing tables to look like they are PF4 when they aren't yet.

**Comment by Ron-Lavi** [comment]
@tbrisker looks like the discussion has evolved in https://community.theforeman.org/t/apply-table-pf4-classname-and-use-pf4-pagination/25640

A few suggestions:
- I can split the pagination refactor into another PR so we can start using the PF4 react component which is also needed for the new breadcrumbs by @MariaAga.

- I think the more complex table is the hosts one, we can remove it from this PR and add only a few specific components, by pointing them to use another class while the original css table classes will stay untouched, by doing so we could iterate a few pages, PR by PR and add it in a secure way.

I also believe the index page template will be a better solution, but it's not that close to get in.

**Comment by Ron-Lavi** [comment]
Yesterday I also raised your concerns in the [UX Interest Group Meeting 18/10/21](https://community.theforeman.org/t/ux-interest-group-meeting-18-10-21/25735?u=laviro)
and we decided that it will be better to invest in the index page template for a better UX.

The idea was to make the UI more consistent and to touch those old pages and update them a bit.
but for a better experience, e.g: using React router and avoiding full page reloads it will be better to use the template.

Overall I am happy this discussion happened and I will pass your inputs to the PF4 team.

We decided that after the index page template will get in if there will still be some leftover tables then we can apply those styles to them.

I am closing this for now and will open a PR with the new PF4 pagination.

**Pull request closed without merging**